addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1593,25 +1597,34 @@ const Wysiwyg = Widget.extend({
             });
         }
     },
-    _processAndApplyColor: function (eventName, color) {
+    _processAndApplyColor: function (eventName, color, previewMode) {
         if (!color) {
             color = 'inherit';
         } else if (!ColorpickerWidget.isCSSColor(color) && !weUtils.isColorGradient(color)) {
             color = (eventName === "foreColor" ? 'text-' : 'bg-') + color;
         }
         const fonts = this.odooEditor.execCommand('applyColor', color, eventName === 'foreColor' ? 'color' : 'backgroundColor', this.lastMediaClicked);
 
-        // Ensure the selection in the fonts tags, otherwise an undetermined
-        // race condition could generate a wrong selection later.
-        const first = fonts[0];
-        const last = fonts[fonts.length - 1];
-
-        const sel = this.odooEditor.document.getSelection();
-        sel.removeAllRanges();
-        const range = new Range();
-        range.setStart(first, 0);
-        range.setEnd(...endPos(last));
-        sel.addRange(range);
+        const didColorTds = fonts.some(font => font.classList.contains('o_selected_td'));
+        if (didColorTds) {
+            const propName = eventName === 'foreColor' ? 'color' : 'background-color';
+            for (const td of fonts) {
+                // Make it important so it has priority over selection color.
+                td.style.setProperty(propName, td.style[propName], previewMode ? 'important' : '');
+            }
+        } else {
+            // Ensure the selection in the fonts tags, otherwise an undetermined
+            // race condition could generate a wrong selection later.
+            const first = fonts[0];
+            const last = fonts[fonts.length - 1];
+
+            const sel = this.odooEditor.document.getSelection();
+            sel.removeAllRanges();
+            const range = new Range();
+            range.setStart(first, 0);
+            range.setEnd(...endPos(last));
+            sel.addRange(range);
+        }
```

Comment:
We definitely need to talk about the interactions between the colorpicker and the tables.

Reply:
Whenever you want.